### PR TITLE
Backport phive to 4.x

### PR DESCRIPTION
**Pull request by markstory**
I like this workflow quite a bit. We should use it in 4.x too. It is faster and more efficient than before.

I'm also trying to use it in CI configuration. If it goes well I can spread this around to the 5.x config repository.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -220,44 +220,42 @@ jobs:
...
+    - name: Install PHP tools with phive.
+      uses: szepeviktor/phive@v1
+      with:
+        home: "${{ runner.temp }}/.phive"
+        binPath: "${{ github.workspace }}/tools/phive"
+        trustGpgKeys: "CF1A108D0E7AE720, 12CE0F1D262429A5"
 
-    - name: Run PHP CodeSniffer
+    - name: Run phpcs
       run: vendor/bin/phpcs --report=checkstyle src/ tests/ | cs2pr
```
Comment (othercorey) [inline .github/workflows/ci.yml:257]: still need phpcs installed.

Comment (markstory) [reply .github/workflows/ci.yml:257]: Oops.

Comment (othercorey) [reply .github/workflows/ci.yml:257]: We could just copy the shared cs-stan workflow from 5.x into a 4.x branch and use it here.

https://github.com/cakephp/.github/blob/5.x/.github/workflows/cs-stan.yml

Comment (markstory) [reply .github/workflows/ci.yml:257]: But that workflow isn't using phive to install tools yet. There are additional copies of tool version numbers in those workflow steps. I'd like to not have to maintain versions in two places.

**Pull request merged**